### ISSUE TITLE
More investigation user side needed. [close]

### ISSUE THREAD

**Issue by steljwagh** [cosmoderp/Marlin#3]
[s]I thought a warning is in place with this version.

Although I love the menu options, this is my favourite f/w for the ender 3 v2. But unfortunately, 2 different users with the samse ce3v2, including me, suffer from severe nozzle crash problems with bltouch.
(It looks like auto leveling works fine, and sometimes it does actually, but sometimes it crashes into the bed hard, bending the whole x gantry.)

I don't know what the reason is, but fairly same config files with other 2.x firmwares work fine. Do not use this firmware with bltouch to prevent damage to your ender 3 v2.
[/s]

**Comment by skaza2125** [comment]
it doesn't crash your nozzle. You probably did not configure it correctly or you did not configure your hardware correctly. Hundreds of users including me are using this firmware as it is one of the only working v2 firmwares. I have used it for over 3 months with no issue.

**Comment by NoTrace-Kai** [comment]
Hi ...
can you perhaps tell us, how the bed leveling with bltouch work on your firmware?
Thanks Kai

**Comment by steljwagh** [comment]
I included both config files, the only files altered for building this firmware. I also used the Advanced menu branch.

With compiling it's succesful, but with these warnings:

```
C:\Users\admin\.platformio\packages\framework-arduinoststm32-maple@2.10000.200103\STM32F1\libraries\STM32ADC\src\utility\util_adc.c:10:30: warning: 'adc_result' initialized and declared 'extern'
 extern volatile unsigned int adc_result = 0;
                              ^~~~~~~~~~
C:\Users\admin\.platformio\packages\framework-arduinoststm32-maple@2.10000.200103\STM32F1\libraries\FreeRTOS701\utility\port.c:178:2: warning: implicit declaration of function 'systick_attach_callback' [-Wimplicit-function-declaration]
  systick_attach_callback(&xPortSysTickHandler);
  ^~~~~~~~~~~~~~~~~~~~~~~
C:\Users\admin\.platformio\packages\framework-arduinoststm32-maple@2.10000.200103\STM32F1\libraries\FreeRTOS701\utility\port.c:285:6: warning: 'prvSetupTimerInterrupt' defined but not used [-Wunused-function]
 void prvSetupTimerInterrupt( void )
      ^~~~~~~~~~~~~~~~~~~~~~
```

The BLTouch is connected to the dedicated BLTouch port on the ender v2 board, it's working fine with other firmwares, but with this firmware the BLTouch is behaving very weird, 3 selftests at startup and random crashes and weird probing pin behaviour (retracting while probing without touching the bed).

**Comment by steljwagh** [comment]
I'm sorry guys for this issue, I compiled again and errors disappeared now, also altered the original config files again with minimal changes and now it's working as designed. I have to do some homework.

**Comment by steljwagh** [comment]
Ok I did my homework, same config files with main marlin bugfix 2.x, everything works fine. Also cosmoderp compiles ok now, but unfortunately is unreliable and feels very unstable.

Using the cosmoderp advanced menu branche, configuring the offset let's crash your nozzle in your bed. Using bltouch on the bltouch port on the 4.2.2 board, not using the z-axis limit switch port, something isn't right with this firmware. Also, watched midnighttech video to verify if I did something wrong and I didn't. 

Next time I try cosmoderp I'll try to log/video the events.

Meanwhile, running on main marlin with same config files, not any issues so far.
